internal/core/resolved.go

```diff
@@ -17,6 +22,50 @@ type ResolvedRoute interface {
 	GetMeta() map[string]string
 	GetName() string
 	GetNamespace() string
+	Marshal() ([]byte, error)
+}
+
+type marshaledRoute struct {
+	Type  ResolvedRouteType
+	Route json.RawMessage
+}
+
+// TODO Is this the best location for this func?
```

Comment:
I feel like we have marshaling structs + funcs spread across the codebase a little too much, but I've left this as something for the final round of cleanup after we've transitioned to the new `Store`

Reply:
It feels like there _should_ be a cleaner way to implement this (don't the plain `Marshal` and `Unmarshal` function names have some magic behavior? Is that maybe only implemented by [protobuf](https://github.com/golang/protobuf/issues/955)?) but I don't have any concrete suggestions on _how_ though, so deferring any recommendation here for now.